src/queries/monthly-delivery-report.sql

```diff
@@ -0,0 +1,246 @@
+/**
+* This query collects all the Monitoring goals used on approved reports within the defined time range.
+*
+* The query results are filterable by the JDI flags. All JDI flags are passed as an array of values
+* The following are the available flags within this script:
+* - jdi.regionIds - one or more values for 1 through 12
+* - jdi.startDate - two dates defining a range for the startDate to be within
+*
+* zero or more JDI flags can be set within the same transaction as the query is executed.
+* The following is an example of how to set a JDI flag:
+* SELECT SET_CONFIG('jdi.regionIds','[11]',TRUE);
+* SELECT SET_CONFIG('jdi.startDate','["2024-04-01","2024-04-30"]',TRUE);
+*/
+WITH
+  reports AS (
+    SELECT
+      a.id,
+      a."userId",
+      a.duration,
+      a."deliveryMethod",
+      a."calculatedStatus",
+      a."startDate"
+    FROM "ActivityReports" a
+    WHERE a."calculatedStatus" = 'approved'
+    -- Filter for regionIds if jdi.regionIds is defined
+    AND (NULLIF(current_setting('jdi.regionIds', true), '') IS NULL
```

Comment:
Not sure if its possible to just have a base filtered table or activity reports using the flags once?

Reply:
I don't understand what you are asking for.